name.go

```diff
@@ -65,12 +66,20 @@ func (n Name) String() string {
 // Object is a local interface representation of the Kubernetes metav1.Object, to avoid dependencies on
 // k8s.io/apimachinery.
 type Object interface {
-	GetClusterName() string
+	GetAnnotations() map[string]string
+	SetAnnotations(annotations map[string]string)
 }
 
 // From returns the logical cluster name for obj.
 func From(obj Object) Name {
-	return Name{obj.GetClusterName()}
+	//TODO: Do we want some sort of error in this library when this is not found?
```

Comment:
I don't think it's necessary. We don't have it today. It would indicate a more fundamental problem with the storage plumbing.